app/models/github_repository.rb

```diff
@@ -42,6 +84,29 @@ def default_branch
     end
   end
 
+  def branch(name, **options)
+    GitHub::Errors.with_error_handling do
+      @client.branch(full_name, name, options)
+    end
+  rescue GitHub::Error
+    {}
```

Comment:
It returns a Hash if it's successful. More specifically though it's a Sawyer::Resource